caas/kubernetes/provider/k8s.go

```diff
@@ -380,6 +374,34 @@ func (k *kubernetesClient) maybeGetVolumeClaimSpec(params volumeParams) (*core.P
 	}, false, nil
 }
 
+func (k *kubernetesClient) ensureStorageClass(cfg *storageConfig) error {
+	// First see if the named storage class exists.
+	storageClasses := k.StorageV1().StorageClasses()
+	_, err := storageClasses.Get(cfg.storageClass, v1.GetOptions{})
+	if err == nil {
+		return nil
+	}
+
+	// If it's not found but there's no provisioner specified, we can't
+	// create it so just return not found.
+	if !k8serrors.IsNotFound(err) || cfg.storageProvisioner == "" {
+		if k8serrors.IsNotFound(err) {
+			return errors.NotFoundf("storage class %v", cfg.storageClass)
+		}
+		return errors.Trace(err)
+	}
```

Comment:
thinking if we could make here flat like: 
```go
if !k8serrors.IsNotFound(err) {
    return errors.Trace(err)
}
// no existing storageclass, so will try to create one
if cfg.storageProvisioner == "" {
    return errors.NotFoundf("storage class %v, but no storage provisioner is available to ensure it.", cfg.storageClass)
}

...
```

Reply:
I've tweaked the implementation. Note that the suggested code isn't quite correct. It needs to be:
if err != nil && cfg.storageProvisioner == "" {